core/src/main/java/org/arquillian/smart/testing/ObjectMapper.java

```diff
@@ -0,0 +1,133 @@
+package org.arquillian.smart.testing;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ObjectMapper {
+
+    public static <T> T mapToObject(Class<T> aClass, Map<String, Object> map) {
+        if (!map.isEmpty()) {
+            T instance;
+            try {
+                instance = aClass.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException("Failed to create new instance of class " + aClass, e);
+            }
+
+            Arrays.stream(aClass.getMethods()).filter(ObjectMapper::isSetter)
+                .forEach(method -> invokeMethodWithMappedValue(method, instance, map));
+
+            return instance;
+        }
+
+        return null;
+    }
+
+    private static <T> void invokeMethodWithMappedValue(Method method, T instance, Map<String, Object> map) {
+        method.setAccessible(true);
+        final String field = method.getName().substring(3);
+        final String property = Character.toLowerCase(field.charAt(0)) + field.substring(1);
+        final Object mappedValue = map.get(property);
+
+        if (mappedValue != null && method.getParameterTypes().length == 1) {
+            try {
+                final Object converted = convert(method, mappedValue);
+                if (converted != null) {
+                    method.invoke(instance, converted);
+                }
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                throw new RuntimeException("Failed to invoke method " + method, e);
+            }
+        }
+    }
+
+    private static Object convert(Method method, Object mappedValue) {
+        Class<?> parameterType = method.getParameterTypes()[0];
+        if (parameterType.isArray()) {
+            return handleArray(parameterType.getComponentType(), mappedValue);
+        } else if (parameterType.isEnum()) {
+            return handleEnum(method, mappedValue);
+        } else if (parameterType.isAssignableFrom(List.class)) {
+            return handleList(method, mappedValue);
+        } else if (!parameterType.isAssignableFrom(Map.class) && Map.class.isAssignableFrom(mappedValue.getClass())) {
```

Comment:
This if should check other configuration objects such as `Report` and `Scm`. Personally, I would create an interface (eg. `ConfigurationSection`) that all configuration classes would implement it... Then you could check if the parameter type is assignable from the interface.